<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find turtle)/script/data/scene/maps/map0.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="true"/>
  <arg name="init_x" default="0"/>
  <arg name="init_y" default="0"/>
  <arg name="init_yaw" default="0"/>

  <group ns="turtlebot3">
  
    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)">
      <param name="frame_id" value="turtlebot3/map"/> 
    </node>

    <!-- AMCL -->
    <!-- <include file="$(find turtlebot3_navigation)/launch/amcl.launch">
      <arg name="initial_pose_x" value="$(arg init_x)" />
      <arg name="initial_pose_y" value="$(arg init_y)" />
      <arg name="initial_pose_a" value="$(arg init_yaw)" />
    </include> -->

    <!-- move_base -->
    <include file="$(find turtlebot3_navigation)/launch/move_base.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="move_forward_only" value="$(arg move_forward_only)"/>
    </include>
  </group>

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  </group>
</launch>
